examples/misc/raptor_master.py

```diff
@@ -276,20 +280,36 @@ def result_cb(self, tasks):
     # insert `n` worker tasks into the agent.  The agent will schedule (place)
     # those workers and execute them.  Insert one smaller worker (see above)
     # NOTE: this assumes a certain worker size / layout
-    print('workers: %d' % n_workers)
+    out('workers: %d' % n_workers)
     descr['ranks']         = nodes_per_worker * cores_per_node
     descr['gpus_per_rank'] = nodes_per_worker * gpus_per_node
-    master.submit_workers(descr=descr, count=n_workers)
+    worker_ids = master.submit_workers(
+                 [rp.TaskDescription(descr) for _ in range(n_workers)])
 
     # wait until `m` of those workers are up
     # This is optional, work requests can be submitted before and will wait in
     # a work queue.
-  # master.wait(count=nworkers)
+    # FIXME
+    master.wait_workers(count=1)
 
+    out('start')
     master.start()
+    out('submit')
     master.submit()
-    master.join()
+
+    # let some time pass for client side tasks to complete
+    time.sleep(60)
+
+    out('stop')
+    # TODO: can be run from thread?
     master.stop()
+    out('join')
+
+    # TODO: worker state callback
+    master.join()
+    out('done')
```

Comment:
Assuming the updated examples have been tested to complete successfully (such as confirming this log message), then this change resolves #2767